src/hugr/validate.rs

```diff
@@ -997,7 +997,11 @@ mod test {
             .map_into()
             .collect_tuple()
             .unwrap();
-
+        // Write Extension annotations into the Hugr while it's still well-formed
+        // enough for us to compute them
```

Comment:
This was the tricky one ;-). I found another way that also works, using `validate_with_extension_closure`; commit 923415479183d54d94d753dc55059aac136ac965 swapped that out for this version, I'd be happy to go with either.

Reply:
(Using that way in the other test!)